test/alternator/test_batch.py

```diff
@@ -375,6 +379,43 @@ def test_batch_write_item_large(test_table_sn):
     assert full_query(test_table_sn, KeyConditionExpression='p=:p', ExpressionAttributeValues={':p': p}
         ) == [{'p': p, 'c': i, 'content': long_content} for i in range(25)]
 
+# Test if client breaking connection during HTTP response
+# streaming doesn't break the server.
+def test_batch_write_item_large_broken_connection(test_table_sn, request, dynamodb):
+    fn_name = sys._getframe().f_code.co_name
+    ses = new_dynamodb_session(request, dynamodb)
+
+    p = random_string()
+    long_content = random_string(100)*500
+    write_reply = test_table_sn.meta.client.batch_write_item(RequestItems = {
+        test_table_sn.name: [{'PutRequest': {'Item': {'p': p, 'c': i, 'content': long_content}}} for i in range(25)],
+    })
+    assert 'UnprocessedItems' in write_reply and write_reply['UnprocessedItems'] == dict()
+
+    read_fun = urllib3.HTTPResponse.read_chunked
+    triggered = False
+    def broken_read_fun(self, amt=None, decode_content=None):
+        ret =  read_fun(self, amt, decode_content)
+        st = traceback.extract_stack()
+        # Try to not disturb other tests if executed in parallel
+        if fn_name in str(st):
+            self._fp.fp.raw.close() # close the socket
+            nonlocal triggered
+            triggered = True
+        return ret
+    urllib3.HTTPResponse.read_chunked = broken_read_fun
+
+    try:
+        # This disruption doesn't always work so we repeat it.
+        for _ in range(1, 20):
```

Comment:
I don't understand what this loop intends to achieve over running once. It checks that on **every iteration**, the disruption happened - that the query must raise HTTPClientError every time, and triggered must be true after each iteration (by the way, after the first time, it just remains true! maybe you meant to set it to false every time?).

What I'm guessing you really wanted to say in the comment above the loop was that before your patch, Scylla only crashed randomly so you needed to run it 20 times to crash more reliably. But this is not what the comment says - it suggests that the "disruption" (the close) doesn't always work, but according to the HTTPClientError you checked below, it seems it does work every time.

Reply:
I suspect following reasons:
- perhaps scylla is sending bytes before we are able to return error in python, for instance it lands in some OS queue or python library queue first, in such case error doesn't always affect the flow 
- it's not a single connection in the driver but a pool (they say default is 10 I think but it's difficult to track in their code as it's very "dynamic", e.g. converts some strings to class names and instantiates it, stuff like that).

Reply:
second point is maybe less relevant here,  as you noticed it client side error always happens

Reply:
although maybe if the connection is closed then reusing it significantly increases crash chance